streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java

```diff
@@ -1304,22 +1307,72 @@ public void shouldDisableMetricCollectionForAllInternalClients() {
     }
 
     @Test
-    public void shouldDisableMetricCollectionOnMainConsumerOnly() {
+    public void shouldThrowConfigExceptionWhenMainConsumerMetricsDisabledStreamsMetricsPushEnabled() {
+        props.put(StreamsConfig.mainConsumerPrefix(ConsumerConfig.ENABLE_METRICS_PUSH_CONFIG), false);
+
+        final Exception exception = assertThrows(ConfigException.class, () -> new StreamsConfig(props));
+
+        assertThat(
+                exception.getMessage(),
+                containsString("KafkaStreams has metrics push enabled" +
+                        " but the main consumer metrics push is disabled. Enable " +
+                        " metrics push for the main consumer")
+        );
+    }
+
+    @Test
+    public void shouldThrowConfigExceptionConsumerMetricsDisabledStreamsMetricsPushEnabled() {
+        props.put(StreamsConfig.consumerPrefix(ConsumerConfig.ENABLE_METRICS_PUSH_CONFIG), false);
         props.put(StreamsConfig.mainConsumerPrefix(ConsumerConfig.ENABLE_METRICS_PUSH_CONFIG), false);
 
+        final Exception exception = assertThrows(ConfigException.class, () -> new StreamsConfig(props));
+
+        assertThat(
+                exception.getMessage(),
+                containsString("KafkaStreams has metrics push enabled" +
+                        " but the consumer clients metrics push is disabled. Enable " +
```

Comment:
Does it add much value to say `consumer clients` for this case instead of `main consumer clients`? Wondering if we can simplify the verification by printing the same error message for both cases. Given that `consumer.xxx` does disable it for the main consumer, printing `main consumer` in the error message would still be correct?